vignettes/advs.Rmd

```diff
@@ -0,0 +1,392 @@
+
+dataset_vignette(advs_param, display_vars = exprs(USUBJID, VSTESTCD, PARAMCD, PARAM, PARCAT1, PARCAT1N))
+```
+
+## Derive Additional Parameters (e.g. `BMI` for `ADVS`) {#derive_param}
+
+In metabolic trials, `BMI` is often calculated at source using collected
+height and weight for screening purposes. But while creating the `ADVS`
+dataset, we re-derive `BMI` from the collected height and weight values.
+This is done to ensure that the `BMI` is calculated consistently across
+all subjects and visits.
+
+In this step, we create parameter `Body Mass Index` (BMI) for `ADVS`
+domain using the `derive_param_bmi()` function. Note that only variables
+specified in the `by_vars` argument will be populated in the newly
+created records. "Also note that if height is collected only once for a
+subject use `constant_by_vars` to specify the function to merge by the
+subject-level variable" Otherwise BMI is only calculated for visits
+where both are collected.
+
+```{r eval=TRUE}
+# Removing BMI collected at source from the dataset
+advs <- advs %>% filter(!VSTESTCD == "BMI")
+
+advs <- derive_param_bmi(
+  advs,
+  by_vars = c(
+    get_admiral_option("subject_keys"),
+    exprs(!!!adsl_vars, VISIT, VISITNUM, ADT, ADY, VSTPT, VSTPTNUM)
+  ),
+  set_values_to = exprs(
+    PARAMCD = "BMI",
+    PARAM = "Body Mass Index (kg/m^2)",
+    PARAMN = 3,
+    PARCAT1 = "Subject Characteristic",
+    PARCAT1N = 1
+  ),
+  get_unit_expr = VSSTRESU,
+  filter = VSSTAT != "NOT DONE" | is.na(VSSTAT),
+  constant_by_vars = exprs(USUBJID)
+)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  arrange(advs, USUBJID, VISITNUM, VSTPTNUM, ADT, PARAMCD),
+  display_vars = exprs(USUBJID, VSTESTCD, PARAMCD, PARAM, VISIT, AVAL),
+  filter = PARAMCD %in% c("BMI")
+)
+```
+
+```{r eval=TRUE, include=FALSE}
+advs <- restrict_derivation(
+  advs,
+  derivation = derive_var_extreme_flag,
+  args = params(
+    by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD)),
+    order = exprs(ADT, VSTPTNUM, VISITNUM),
+    new_var = ABLFL,
+    mode = "last"
+  ),
+  filter = (!is.na(AVAL) & ADT <= TRTSDT)
+)
+
+advs <- derive_var_base(
+  advs,
+  by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD)),
+  source_var = AVAL,
+  new_var = BASE
+)
+
+advs <- derive_var_chg(advs)
+
+advs <- derive_var_pchg(advs)
+```
+
+In the following sections, we will explore some of the most common
+endpoints typically observed in metabolic trials.
+
+One such endpoint is the improvement in weight category from baseline to
+the end of treatment, which is often assessed using Body Mass Index
+(`BMI`). To capture this, we will derive variables such as `AVALCATy`
+and `BASECATy`, as detailed in the subsequent section.
+
+Additionally, the achievement of weight reduction thresholds, such as ≥
+5%, ≥ 10%, or ≥ 15% from baseline to end of treatment or certain visit,
+is a common endpoint in obesity trials. To accommodate these criteria,
```

Comment:
Are we referring to metabolic or obesity trials? Are they synonymous? Maybe we could standardise the language?